packages/sdk/src/chains/evm.ts

```diff
@@ -118,16 +118,22 @@ export class EvmChain implements IChain {
 	}
 
 	/**
-	 * Queries the proof of the requests.
-	 * @param {HexString[]} requests - The requests to query.
+	 * Queries the proof of the commitments.
+	 * @param {HexString[]} commitments - The commitments to query.
 	 * @param {string} counterparty - The counterparty address.
+	 * @param {boolean} isRequest - Whether the commitments are requests or responses.
 	 * @param {bigint} [at] - The block number to query at.
 	 * @returns {Promise<HexString>} The proof.
 	 */
-	async queryRequestsProof(requests: HexString[], counterparty: string, at?: bigint): Promise<HexString> {
+	async queryProof(
+		commitments: HexString[],
+		counterparty: string,
+		isRequest: boolean,
+		at?: bigint,
```

Comment:
```suggestion
		message: IMessage,
		counterparty: string,
		at?: bigint,
```

Maybe instead take the enum directly ?

```ts
type IMessage = { Requests: string[] } |  { Responses: string[] }
```